pagefaultgames/pokerogue
# [Move] Implement Fling

## What are the changes?

The user can now utilize the move fling. 

## Why am I doing these changes?

It is an important move in the game.



Greater diversity of available moves, which enhances user experience.

## What did change?


The move fling is now fully functional.

### Screenshots/Videos

https://github.com/pagefaultgames/pokerogue/assets/128145955/7281d3bf-87da-410e-8d2c-a5aac97c5b5c

## How to test the changes?




I made use of src/overrides.ts to test different items. I did not introduce automated tests. 
I carried out tests on various items and confirmed functionality. If there is an issue, it may be in 
determining which items are able to be flung and which are not. You may want to check my arrays
to see if the items are appropriately sorted. You can also go through the code and try to see
if the logic makes sense. I left behind some comm

My Discord username is bgjormungandr, I'm on the pokerogue server. Feel free to contact me there if needed.

## Checklist
- [ yes] There is no overlap with another PR?
- [ yes] The PR is self-contained and cannot be split into smaller PRs?
- [ yes] Have I provided a clear explanation of the changes?
- [ yes] Have I tested the changes (manually)?
    - [did not implement tests] Are all unit tests still passing? (`npm run test`)
- [ yes] Are the changes visual?
  - [yes ] Have I provided screenshots/videos of the changes?
